preview-env/create/action.yml

```diff
@@ -68,10 +68,38 @@ runs:
       ref: ${{ inputs.revision }}
   - name: Deploy
     shell: bash
-    run: >-
-      argocd app create ${{ inputs.argocd_arguments }}
+    run: |
+      max_retries=3
+      for attempt in $(seq "$max_retries"); do
+          argocd app create ${{ inputs.argocd_arguments }}
+          if [ $? -eq 0 ]; then
+            echo "Creating Argo CD app with name ${{ inputs.app_name }}"
+            sleep 60
 
-      argocd app sync ${{ inputs.app_name }} --async --force || true
+            # Fetch app information to make sure the app is registered
+            app_info=$(argocd app get ${{ inputs.app_name }} --output json 2>/dev/null)
+            get_status_status=$?
+            if [ $get_status_status -eq 0 ] && [ -n "$app_info" ]; then
```

Comment:
since you're immediately using `get_status_status`, you can just skip the variable assignment.